### PR TITLE
[Java][client] Fix feign classcastexception when getting headers

### PR DESCRIPTION
**Pull request by robertdanci**
This affects the feign apis that return _ApiResponse_ which contains extra request info such as headers and status.
The current implemenation holds the headers in a Map<String, List<String>> but feign uses Map<String, Collection<String>>. Since the headers are copied as they are in _ApiResponseDecoder.java_  when one tries to get them one gets a ClassCastException. That is because Feign wrapps the headers in an unmodifiableCollection, see [here](https://github.com/OpenFeign/feign/blob/master/core/src/main/java/feign/Util.java#L398). Regardless of the underlying implementation the correct type is Map<String, Collection<String>> but since the _ApiResponseDecoder_ uses raw types, this type missmatch is hidden. This has also been addressed. See [here](https://github.com/OpenFeign/feign/blob/master/core/src/main/java/feign/Response.java#L30) Feign's response for context.

I also deleted feign10x sample folder since it seems it not in use anymore. The generator only supports Feign 10.x+ so there no need for something version specific.

@bbdouglas @sreeshas @jfiala @lukoyanov @cbornet @jeff9finger @karismann @Zomzog @lwlee2608 @martin-mfg 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: samples/client/petstore/java/feign-no-nullable/src/main/java/org/openapitools/client/model/HttpResponse.java [hunk 1]
```diff
@@ -1,31 +0,0 @@
...
-
-public class HttpResponse<T>{
-
-  private Map<String, Collection<String>> headers;
-
-  private T body;
-
-  private int status;
-
-  public HttpResponse(Map<String, Collection<String>> headers, T body, int status) {
```
Comment (robertdanci) [inline samples/client/petstore/java/feign-no-nullable/src/main/java/org/openapitools/client/model/HttpResponse.java:14]: No longer used, replaced by ApiResponse

File: samples/client/petstore/java/feign-no-nullable/src/main/java/org/openapitools/client/JacksonResponseDecoder.java [hunk 2]
```diff
@@ -1,38 +0,0 @@
...
-import java.io.IOException;
-import java.lang.reflect.ParameterizedType;
-import java.lang.reflect.Type;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Map;
-
-import org.openapitools.client.model.HttpResponse;
-
-public class JacksonResponseDecoder extends JacksonDecoder {
```
Comment (robertdanci) [inline samples/client/petstore/java/feign-no-nullable/src/main/java/org/openapitools/client/JacksonResponseDecoder.java:17]: No longer used, replaced by ApiResponseDecoder

File: .gitignore [hunk 3]
```diff
@@ -49,8 +49,6 @@ nb-configuration.xml
 /target
 /generated-files
 test-output/
-nbactions.xml
```
Comment (robertdanci) [inline .gitignore:52]: I removed a bunch of duplicates

**Comment by wing328** [comment]
thanks for the PR with the detailed explanation.

lgtm

**Pull request merged**